File: algorithms/Monotonic Array/Monotonic Array.js
var isMonotonic = function(nums) {
    let increasingCount = 0;
    let decreasingCount = 0;
    for(let i = 1; i < nums.length; i++){
        if(nums[i] > nums[i-1]){
            increasingCount++;
        }else if(nums[i] < nums[i-1]){
            decreasingCount++;
        }
    }
    
    return !(increasingCount && decreasingCount);
};

File: algorithms/M/Monotonic Array/Monotonic Array.py
class Solution:
    def isMonotonic(self, nums: List[int]) -> bool:
        counter = 0
        for i in range(len(nums) - 1):
            if nums[i] >= nums[i + 1]:
                counter += 1
        if counter == len(nums) - 1:
            return True
        counter = 0
        for i in range(len(nums) - 1):
            if nums[i] <= nums[i + 1]:
                counter += 1
        if counter == len(nums) - 1:
            return True
        return False
